src/cli/self_update/unix.rs

```diff
@@ -57,66 +57,50 @@ pub fn complete_windows_uninstall() -> Result<utils::ExitCode> {
     panic!("stop doing that")
 }
 
-pub fn do_remove_from_path(methods: &[PathUpdateMethod]) -> Result<()> {
-    for method in methods {
-        if let PathUpdateMethod::RcFile(ref rcpath) = *method {
-            let file = utils::read_file("rcfile", rcpath)?;
-            let addition = format!("\n{}\n", shell_export_string()?);
+pub fn do_remove_from_path() -> Result<()> {
+    for sh in shell::get_available_shells() {
+        let source_cmd = format!("\n{}\n", sh.source_string()?);
+        let source_bytes = source_cmd.into_bytes();
```

Comment:
this probably needs to be bytes the whole way through - because of the issue I mentioned with potential non-unicode translatable paths.